### ISSUE TITLE
Fuzz error

### ISSUE THREAD

**Issue by redboltz** [msgpack/msgpack-c#969]
I got the follwing fuzz error report:

----

New issue 37845 by ClusterFuzz-External: msgpack-c: Fuzzing build failure
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=37845

The last 2 builds for msgpack-c have been failing.
Build log: https://oss-fuzz-build-logs.storage.googleapis.com/log-812fd458-15e4-48d0-a663-2fb92f60a32f.txt
Build type: fuzzing

To reproduce locally, please see: https://google.github.io/oss-fuzz/advanced-topics/reproducing#reproducing-build-failures

This bug tracker is not being monitored by OSS-Fuzz team. If you have any questions, please create an issue at https://github.com/google/oss-fuzz/issues/new.

**This bug will be automatically closed within a day once it is fixed.**

----

It seems that undefined reference LLVMFuzzerTestOneInput causes the link error:

```
Starting Step #3 - "compile-afl-address-x86_64"
Step #3 - "compile-afl-address-x86_64": Already have image: gcr.io/oss-fuzz/msgpack-c
Step #3 - "compile-afl-address-x86_64": ---------------------------------------------------------------
Step #3 - "compile-afl-address-x86_64": Copying precompiled afl++
Step #3 - "compile-afl-address-x86_64": 
Step #3 - "compile-afl-address-x86_64": afl++ target compilation setup:
Step #3 - "compile-afl-address-x86_64": AFL_LLVM_MODE_WORKAROUND=0
Step #3 - "compile-afl-address-x86_64": AFL_ENABLE_DICTIONARY=0
Step #3 - "compile-afl-address-x86_64": AFL_LLVM_INSTRUMENT=CLASSIC,CTX-2
Step #3 - "compile-afl-address-x86_64": AFL_ENABLE_CMPLOG=0
Step #3 - "compile-afl-address-x86_64": AFL_LAF_CHANCE=30
Step #3 - "compile-afl-address-x86_64": AFL_QUIET=1
Step #3 - "compile-afl-address-x86_64": 
Step #3 - "compile-afl-address-x86_64":  done.
Step #3 - "compile-afl-address-x86_64": ---------------------------------------------------------------
Step #3 - "compile-afl-address-x86_64": CC=/src/aflplusplus/afl-clang-fast
Step #3 - "compile-afl-address-x86_64": CXX=/src/aflplusplus/afl-clang-fast++
Step #3 - "compile-afl-address-x86_64": CFLAGS=-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope 
Step #3 - "compile-afl-address-x86_64": CXXFLAGS=-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope  -stdlib=libc++
Step #3 - "compile-afl-address-x86_64": RUSTFLAGS=--cfg fuzzing -Zsanitizer=address -Cdebuginfo=1 -Cforce-frame-pointers
Step #3 - "compile-afl-address-x86_64": ---------------------------------------------------------------
Step #3 - "compile-afl-address-x86_64": + for f in $SRC/msgpack-c/fuzz/*_fuzzer.cpp
Step #3 - "compile-afl-address-x86_64": ++ basename /src/msgpack-c/fuzz/unpack_pack_fuzzer.cpp .cpp
Step #3 - "compile-afl-address-x86_64": + fuzzer=unpack_pack_fuzzer
Step #3 - "compile-afl-address-x86_64": + /src/aflplusplus/afl-clang-fast++ -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -stdlib=libc++ -std=c++11 -Iinclude -I/src/msgpack-c/include /src/msgpack-c/fuzz/unpack_pack_fuzzer.cpp -o /workspace/out/afl-address-x86_64/unpack_pack_fuzzer /usr/lib/libFuzzingEngine.a
Step #3 - "compile-afl-address-x86_64": /usr/bin/ld: /usr/lib/libFuzzingEngine.a(aflpp_driver.o): in function `main':
Step #3 - "compile-afl-address-x86_64": /src/aflplusplus/utils/aflpp_driver/aflpp_driver.c:285: undefined reference to `LLVMFuzzerTestOneInput'
Step #3 - "compile-afl-address-x86_64": /usr/bin/ld: /src/aflplusplus/utils/aflpp_driver/aflpp_driver.c:303: undefined reference to `LLVMFuzzerTestOneInput'
Step #3 - "compile-afl-address-x86_64": /usr/bin/ld: /usr/lib/libFuzzingEngine.a(aflpp_driver.o): in function `ExecuteFilesOnyByOne':
Step #3 - "compile-afl-address-x86_64": /src/aflplusplus/utils/aflpp_driver/aflpp_driver.c:191: undefined reference to `LLVMFuzzerTestOneInput'
Step #3 - "compile-afl-address-x86_64": clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
Step #3 - "compile-afl-address-x86_64": ********************************************************************************
Step #3 - "compile-afl-address-x86_64": Failed to build.
Step #3 - "compile-afl-address-x86_64": To reproduce, run:
Step #3 - "compile-afl-address-x86_64": python infra/helper.py build_image msgpack-c
Step #3 - "compile-afl-address-x86_64": python infra/helper.py build_fuzzers --sanitizer address --engine afl --architecture x86_64 msgpack-c
Step #3 - "compile-afl-address-x86_64": ********************************************************************************
Finished Step #3 - "compile-afl-address-x86_64"
ERROR
ERROR: build step 3 "gcr.io/oss-fuzz/msgpack-c" failed: step exited with non-zero status: 1
```

**Comment by redboltz** [comment]
@kovdan01 , perhaps it could caused by the build tree updating. Did you update something on fuzz link ?

**Comment by redboltz** [comment]
@derwolfe, do you know what is a trigger to send error report? 
Does fuzz checking run only on release, or on merge into master branch?

**Comment by kovdan01** [comment]
@redboltz It seems that the problem is with oss-fuzz project but not with msgpack-c

```
Step #3 - "compile-afl-address-x86_64": /src/aflplusplus/utils/aflpp_driver/aflpp_driver.c:285: undefined reference to `LLVMFuzzerTestOneInput'
```

It is expected since I renamed `LLVMFuzzerTestOneInput` to `FuzzerTestOneInput` in #951 because there is **nothing** LLVM-specific there.

I'll submit a PR with a fix to oss-fuzz project.

**Comment by kovdan01** [comment]
@redboltz Update: it seems that naming the function `LLVMFuzzerTestOneInput` is a convention (google's fuzz engine wants to use specifically this function). I'll submit a PR renaming it back to `LLVMFuzzerTestOneInput` and adding a comment to make it clear for those who read the code.

**Comment by kovdan01** [comment]
@redboltz Submitted #972

**Comment by redboltz** [comment]
> @redboltz Update: it seems that naming the function `LLVMFuzzerTestOneInput` is a convention (google's fuzz engine wants to use specifically this function). I'll submit a PR renaming it back to `LLVMFuzzerTestOneInput` and adding a comment to make it clear for those who read the code.

@kovdan01 thank you! 
I found a document: https://llvm.org/docs/LibFuzzer.html

**Comment by redboltz** [comment]
Fixed by #972.

**Comment by redboltz** [comment]
I got the following notify mail. I will update pre 4.0.2 to official.

Updates:
	Status: Verified

Comment #1 on issue 37845 by ClusterFuzz-External: msgpack-c: Fuzzing build failure
https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=37845#c1

The latest build has succeeded, closing this issue.
